### PR TITLE
Remove KOKKOS_CXX_STANDARD mentioning from BUILD.md

### PR DESCRIPTION
**Pull request by crtrott**

### CODE REVIEW

**Review by janciesko — APPROVED** [review]

File: BUILD.md [hunk 1]
```diff
@@ -52,6 +52,10 @@ There are numerous device backends, options, and architecture-specific optimizat
 ````
 which activates the OpenMP backend. All of the options controlling device backends, options, architectures, and third-party libraries (TPLs) are given below.
 
+Kokkos requires as a minimum C++17, however C++20 and C++23 are supported depending on the compiler.
```
Comment (dalg24) [inline BUILD.md:55]: Sounds good but what does "C++20 and C++23 are supported depending on the compiler" actually mean?

Comment (crtrott) [reply BUILD.md:55]: whether the compiler supports it?

**Comment by ajpowelsnl** [comment]
@crtrott - can we close?

**Comment by crtrott** [comment]
It is merged so that is the equivalent of closing ...

**Pull request merged**